contracts/clients/src/BlsWalletWrapper.ts

```diff
@@ -81,31 +108,33 @@ export default class BlsWalletWrapper {
    */
   static async connect(
     privateKey: string,
-    verificationGatewayAddress: string,
-    provider: ethers.providers.Provider,
+    verificationGateway: VerificationGateway,
   ): Promise<BlsWalletWrapper> {
-    const network = await provider.getNetwork();
-
     const blsWalletSigner = await initBlsWalletSigner({
-      chainId: network.chainId,
+      chainId: (await verificationGateway.provider.getNetwork()).chainId,
     });
 
-    const contractAddress = await BlsWalletWrapper.Address(
+    const blsWalletWrapper = new BlsWalletWrapper(
+      blsWalletSigner,
       privateKey,
-      verificationGatewayAddress,
-      provider,
+      verificationGateway,
+      await BlsWalletWrapper.BLSWallet(privateKey, verificationGateway),
     );
 
-    const walletContract = BLSWallet__factory.connect(
-      contractAddress,
-      provider,
+    await blsWalletWrapper.syncWallet(verificationGateway);
```

Comment:
I think the `syncWallet` here is redundant since the latest address information is gathered above during `await BlsWalletWrapper.BLSWallet(privateKey, verificationGateway)`?

Reply:
This is currently only being used in the `should recover before bls key update` contracts test when recovering to a different key, so this still may be needed during the recovery workflow to keep non-async properties of `BlsWalletWrapper` up to date with their on-chain values.

Reply:
Confirmed, removing that `syncWallet` call fails the test `Error: VM Exception while processing transaction: reverted with reason string 'VG: Sig not verified'`